.github/workflows/test.windows.yml

```diff
@@ -15,6 +15,11 @@ jobs:
           go-version: 1.23.x
           cache: false
 
+      - name: Set environment variables
+        run: |
+          echo "GOOS=$(go env GOOS)" >> $GITHUB_ENV
```

Comment:
It doesn't work for windows powershell terminal, fix it.
refer: [Setting an environment variable #PowerShell](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions?tool=powershell#setting-an-environment-variable)

Reply:
Fixed, used `$env:GITHUB_ENV` instead of `$GITHUB_ENV` for pwsh in windows workflow.

In pwsh syntax, the dollar sign `$` indicates a variable, and the drive name `env:` indicates an environment variable followed by the variable name `GITHUB_ENV`.

refer to [Microsoft PowerShell Docs #about_Environment_Variables](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.5#use-the-variable-syntax)